yourbench/utils/inference_engine.py

```diff
@@ -88,8 +88,8 @@ async def _get_response(model: Model, inference_call: InferenceCall) -> str:
 
     # Safe-guarding in case the response is missing .choices
     if not response or not response.choices:
-        logger.warning("Empty response or missing .choices from model {}", model.model_name)
-        return ""
+        logger.error("Empty response or missing .choices from model {}", model.model_name)
+        raise Exception("Failed Inference")
```

Comment:
Where is this error caught for a retry?

Reply:
It's actually caught in the outer function from where it's called

Reply:
https://github.com/huggingface/yourbench/blob/eb50e1fff84849e19ecb62b45a25ed6afed46a2e/yourbench/utils/inference_engine.py#L109C1-L110C1